<launch>

    <param name="scan_line" type="int" value="16" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points -->
    <param name="minimum_range" type="double" value="0.3"/>

    <param name="mapping_line_resolution" type="double" value="0.2"/>
    <param name="mapping_plane_resolution" type="double" value="0.4"/>
    <param name="globalMapFile" type="string" value="/home/xl/Desktop/ht_office/opt/feature_map.pcd"/>

    <param name="init_x" type="double" value="0"/>
    <param name="init_y" type="double" value="0"/>
    <param name="init_z" type="double" value="0"/>

    <param name="init_roll" type="double" value="0.0"/>
    <param name="init_pitch" type="double" value="0.0"/>
    <param name="init_yaw" type="double" value="0.0"/>

    <node pkg="aloam_velodyne" type="aimageProjection" name="aimageProjection" output="screen" />

    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen" />

    <node pkg="aloam_velodyne" type="scan_2_scan" name="scan_2_scan" output="screen" />

    <node pkg="aloam_velodyne" type="lidarMapLocalization" name="lidarMapLocalization" output="screen" />

    <node pkg="aloam_velodyne" type="transformFusion" name="transformFusion" output="screen" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/lidar_localization.rviz" />
    </group>

    <remap from="/sensor/velodyne16/back/PointCloud2" to="/velodyne_points" />
    <node pkg="rosbag" type="play" name="playe" output="screen" args="--clock /home/xl/projects/ws_aloam/hengtong_office_1109.bag" />
</launch>
